pagopa/io-app
# feat: [IOBP-637,IOBP-440] New transactions list and details with biz events integration

## Short description
This PR implements the new transaction list and transaction details powered by biz-events by managing more information such as logo, payment method used, cart type transactions and so on.

## List of changes proposed in this pull request
- Upgraded the `io-app-design-system` library to the latest version in order to support the `ListItemInfo` with a payment method logo.
- Added a new feature-folder within the “payments” folder;
- Added sagas, reducers, store, screens;
- Added selectors for getting latest transactions;
- Added a transaction list with server-side pagination;
- Added biz-events API definitions reference;
<details><summary>Why is a new feature-folder being created? </summary>
<p>
Because the old transactions feature will still have to be available since in addition to biz-events in an early transition there will also be dated transactions
</p>
</details> 
 

## How to test
- Checkout this PR on the io-dev-api-server: https://github.com/pagopa/io-dev-api-server/pull/369
- Now You can test this feature on the main landing page of the new payment section. You should be able to initially see the last 10 transactions made. In the upper right-hand corner, you can see all transactions performed via the “See All” call-to-action, which will show a list of transactions paged 10 per page.
If you tap on a transaction, you can access its detail

## Preview

https://github.com/pagopa/io-app/assets/34343582/9a089ac1-7e25-4d5f-b161-4dc6b82f414c
